src/backend/CryptoDoge.DAL.UnitTests/AuthRepositoryTests.cs

```diff
@@ -244,6 +228,98 @@ public async Task RegisterAsync_CouldntCreateTheUser()
 
 				Assert.Fail();
 			}
+		}
+
+		[Test]
+		public async Task RegisterAsync_AsAdmin()
+		{
+			var roles = new IdentityRole[]
+			{
+				new IdentityRole("Admin"),
+				new IdentityRole("User"),
+			};
+
+			User user;
+			using (var context = new ApplicationDbContext(options))
+			{
+				user = new User
+				{
+					Id = "adminId",
+					UserName = "admin",
+					NormalizedUserName = "admin",
+					Email = "admin@mail.com",
+					NormalizedEmail = "admin@mail.com",
+					EmailConfirmed = false,
+					RefreshToken = "adminRefreshToken",
+				};
+
+				var passwordHasher = new PasswordHasher<User>();
+				user.PasswordHash = passwordHasher.HashPassword(user, "Password");
+
+				await context.Users.AddAsync(user);
+				await context.Roles.AddRangeAsync(roles);
+				await context.SaveChangesAsync();
+			}
+
+			userManager.Setup(r => r.AddToRoleAsync(It.IsAny<User>(), "Admin")).Returns(Task.FromResult(IdentityResult.Success));
+			userManager.Setup(r => r.GetUsersInRoleAsync("Admin")).Returns(Task.FromResult(new List<User> { user } as IList<User>));
+			await userManager.Object.AddToRoleAsync(user, "Admin");
+
+			using (var context = new ApplicationDbContext(options))
+			{
+				var usersInAdminRoles = await userManager.Object.GetUsersInRoleAsync("Admin");
+				Assert.IsTrue(usersInAdminRoles.Any(x => x.UserName == user.UserName));
+
+				context.Users.Remove(user);
+				context.Roles.RemoveRange(roles);
+				await context.SaveChangesAsync();
+			}
+		}
+
+		[Test]
+		public async Task RegisterAsync_AsUser()

```

Comment:
I think RegisterAsync_AsUser and RegisterAsync_AsAdmin tests can be merged. There is an attribute called TestCase which you can use to parameterize tests. https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html